pkg/dependencies/factory.go

```diff
@@ -167,6 +168,20 @@ func (f *Factory) WithProxyConfiguration(proxyConfig map[string]string) *Factory
 	return f
 }
 
+func (f *Factory) WithProxyConfigurationFromEnv() *Factory {
+	
+	if f.proxyConfiguration == nil {
+		proxyConfig := map[string]string{
+			"HTTP_PROXY":  os.Getenv("HTTP_PROXY"),
```

Comment:
Should these all be constants?

Reply:
Looking at this, I wonder if we should make this map part of the config in the `config` package

Reply:
Yea maybe this isn't something we need to keep in factory and have to specify for all dependencies where we want to share this logic